source/boundary_composition/function.cc

```diff
@@ -0,0 +1,205 @@
+/*
+  Copyright (C) 2011 - 2017 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/boundary_composition/function.h>
+#include <aspect/utilities.h>
+#include <aspect/global.h>
+#include <deal.II/base/signaling_nan.h>
+
+namespace aspect
+{
+  namespace BoundaryComposition
+  {
+
+    template <int dim>
+    double
+    Function<dim>::
+    boundary_composition (const types::boundary_id /*boundary_indicator*/,
+                          const Point<dim> &position, const unsigned int n_comp) const
```

Comment:
please rename the last argument (also in the declaration, if applicable). The original declataion in the base class looks like this:
```

        virtual
        double
        boundary_composition (const types::boundary_id boundary_indicator,
                              const Point<dim> &position,
                              const unsigned int compositional_field) const;

```